next/schema-generator/update.ts

```diff
@@ -0,0 +1,318 @@
+
+  if (!token) throw new Error('Token not found for environment')
+
+  const response = await prompts({
+    onState: (state) => {
+      if (state.aborted) {
+        process.nextTick(() => {
+          process.exit(0)
+        })
+      }
+    },
+    type: 'confirm',
+    name: 'value',
+    message: `Re-generate all schema files first ? The 'gestor' will be set according to environment selected in previous step.
+${chalk.redBright('This will delete the entire dist directory!')}
+${chalk.reset.italic(
+  'This step requires that you have @bratislava/jsxt installed globally. If you need just schema.json and uiSchema.json for all forms without the xml transforms you may use ',
+)}${chalk.reset('yarn generate:all')}
+`,
+  })
+  if (response.value) {
+    rimrafSync('./dist', {}) // Empty options object as second argument
+    await execSync('yarn generate:all')
+    const subdirs = fs.readdirSync('./dist')
+    const gestor = isProduction ? 'Martin Pinter' : 'Tisici Janko'
+    console.log(`Generating with gestor ${chalk.bold(gestor)}:`)
+    for (const subdir of subdirs) {
+      console.log(subdir)
+      await execSync(`cd ./dist/${subdir} && jsxt pack -j schema.json --gestor="${gestor}"`)
+    }
+  } else {
+    console.log('Note that if the schemas are not present in dist directory the script will fail.')
+  }
+
+  const files = fs.readdirSync('./dist')
+  const choices = files.map((file) => ({ title: file, value: file }))
+  const chosenSlugsResponse = await prompts({
+    onState: (state) => {
+      if (state.aborted) {
+        process.nextTick(() => {
+          process.exit(0)
+        })
+      }
+    },
+    type: 'multiselect',
+    name: 'selectedSlugs',
+    message: 'Select the schemas you want to use',
+    choices: choices,
+  })
+
+  // each of the prompts needs an onState handler to exit on ctrl+c, otherwise it would be understood as empty answer
+  const actionResponse = await prompts({
+    onState: (state) => {
+      if (state.aborted) {
+        process.nextTick(() => {
+          process.exit(0)
+        })
+      }
+    },
+    type: 'select',
+    name: 'action',
+    message: 'Do you want to patch the existing schema version or upgrade to a new one?',
+    choices: [
+      { title: 'Patch existing', value: 'PATCH' },
+      { title: 'Upgrade to new', value: 'POST' },
+    ],
+  })
+
+  for (const slug of chosenSlugsResponse.selectedSlugs) {
+    const schemaVersionsUrl = `${environment}/schemas/schema-versions?onlyLatest=true&slug=${slug}&currentPage=1&pagination=10`
+    console.log(`Fetching from: ${schemaVersionsUrl}`)
+    const schemaVersionsResponse = await fetch(schemaVersionsUrl, {
+      method: 'GET',
+      headers: {
+        accept: 'application/json',
+      },
+    })
+
+    console.log(
+      `Response status: ${schemaVersionsResponse.status} ${schemaVersionsResponse.statusText}`,
+    )
+
+    const data: any = await schemaVersionsResponse.json()
```

Comment:
why any?

Reply:
cause I'm lazy, this is an internal tool and json() returns unknown

Reply:
ah yea, it doesn't use the api, then it makes sense